### DISCUSSION TITLE
SEQURE SI012 (TS100 clone w/ USB-PD)

### DISCUSSION THREAD

**Discussion by Zipties** [Ralim/IronOS#1301] [Q&A]
I picked this up the other day and I think it should run IronOS alright being it is mostly a clone of the ts100, but having USB-PD I am not sure which firmware package I should go with.

**Comment by Ralim** [comment]
Hello, If sequre release schematics it would be interesting to look at support.

It looks like an expensive copy of the pinecil at first glance but if hardware is similar to others support should be doable.

**Reply by Zipties** [reply]
> It looks like an expensive copy of the pinecil at first glance but if hardware is similar to others support should be doable.

Eek, it sure does. I was looking at their ts100 clone and grabbed this one because it had PD and the same specs. 

That’s embarrassing 🤦‍♂️

**Comment by Egrchs** [comment]
Hello! I would like to know if you managed to install the IRON OS firmware on SEQURE SI 012? Thanks!

**Comment by mr-sm1th** [comment]
a friend just got screwed by this company,
there are 2 SI012 versions, old and new.
new version identifies as a usb mass-storage on the pc,
old version identifies as a usb serial port.
the scumbags - i mean company says the older version cant be upgraded and is using the "final" software version.
this iron running old bootloader/firmware or whatever was only delivered 2 days ago!

so a question to anybody who has a "SI012", does it identify as mass-storage or serial? and what is the cpu part number in it?
i want to find out if they changed the hardware or only the firmware.

if your thinking of buying one of these - dont!

**Reply by mr-sm1th** [reply]
just got sent some pcb pics.
i hope it has a second microcontroller under the oled,
because in the pictures this crap has an Atmel atmega328p in it!!!!!
incase anybody asks, the pcb is black. i have been told some have a red pcb so i dont know if that relates.

**Reply by River-Mochi** [reply]
> old version identifies as a usb serial port.

usb  serial port is ok  for updating.  This is how Pinecil works.    Pinecil V2   identifies in Windows as  Serial COM port  when you hold down the [-]   button before  plugging into the PC. 
Pinecil is not like   ts100,   Pinecil will also never show up as a USB  mass storage device and that is done on purpose.

On  newest V2 pinecil,  We simply  hold down [-] button,  Plug it into USB-C cable.  The windows PC for example device manager  adds it as a new Serial Port COMM.   then  we run  an updater from the chip manufacturer  or   convenient Community made  flasher that is CLI .
grab  IronOS from here,  and it's done.

Therefore  you might Still be able to upgrade  your  Sequire S1012.   
1.  Ask  them  what buttons to press to enter flashing / Update mode.  how long and procedure basically. 
2.   ask them for a  link to schematic,  tell them that is the only way that open source  Ralim's IronOS  could  start looking at it for possible  use of IronOS   on the device.
3. find out from them  EXACTLY  which  MCU chip  is used on it. sometimes you can use a chip flasher from the manufacturer of the chip to flash it with IronOS also.

**Reply by mr-sm1th** [reply]
they dont care, they offered a 10% refund because it cant be updated as described and left it at that.
the usb-serial shows up without pressing buttons and the cpu is an atmega328p with a huge 32k of flash and i actually doubt it's connected to the usb directly because running a soft usb-core takes a lot from the cpu.
*maybe* it has a WCH usb-serial chip, but even then there is no known upload protocol to use.
*if* software was written for it, you would have to use something like a USBasp to write a new bootloader first - maybe the arduino one would be a good choice.

**Reply by Blake1777** [reply]
If anyone sees this thread regarding sequre si0120 soldering iron you want the pro version not the original and I'm not sure the custom firmware will work as its not really a ts100 clone the sequre sq-001 was an exact ts100 clone that works with the ralim firmware and the custom logo editor and I know from my own experience but the one difference with sequre si0120 is it has a beeper inside so if you changed firmware you would need to add this and other features. As a soldering iron it's well worth the purchase even in 2024 (no firmware update)

**Comment by Dmitriy369856** [comment]
SI012  
![3](https://user-images.githubusercontent.com/120063968/206312222-76419b2a-5830-4d52-a729-d2a8f34534ed.jpg)
![123](https://user-images.githubusercontent.com/120063968/206316295-f03677ec-43b6-4461-b585-50f306767eb6.jpg)

Microcontroller  CKS32F103C8T6
![4](https://user-images.githubusercontent.com/120063968/206312271-8c8b381c-5654-4f2b-8340-01e96a62a521.jpg)

Identification is possible only with a clamped button A or B with a subsequent connection to a PC
Pressed button A, connection to PC
![111](https://user-images.githubusercontent.com/120063968/206312758-ab2e96d1-b91d-49b6-88c8-2e9b20829b76.jpg)

Pressed button B, connection to PC
![222](https://user-images.githubusercontent.com/120063968/206312775-6149adec-c965-4080-976f-f2b41ee03913.jpg)

**Reply by mr-sm1th** [reply]
thats the newer revision.
thanks for posting.

**Reply by River-Mochi** [reply]
Who makes that MCU?

**Reply by mr-sm1th** [reply]
looks like a clone of an ST 32f103 with 64k of flash

**Reply by Dmitriy369856** [reply]
Yes, CKS32F103C8T6 is the Chinese ? analogue STM32F103C8T6 only with 64KB flash memory, and not 128kb like most STM
CKS32F103C8T6 Package LQFP-48 ARM Cortex-M3 72MHz Flash Memory: 64K@x8bit RAM: 20KB MCU (MCU/MPU/SOC)
there is a little more here, if interesting   https://hackaday.com/2020/10/22/stm32-clones-the-good-the-bad-and-the-ugly/

**Reply by Dmitriy369856** [reply]
An attempt to update the firmware on TS100. The sting of the soldering iron does not heat up. The screen resolution does not correspond
![1](https://user-images.githubusercontent.com/120063968/206785340-d66f2daa-8a5c-49de-b83d-60900b686508.jpg)

**Reply by PhilipWhiteside** [reply]
> An attempt to update the firmware on TS100. The sting of the soldering iron does not heat up. The screen resolution does not correspond

Same result for me. I have a USB power monitor which showed 0W too. With the Sequre OS it pulls 25W when heating up. 

The text appears to be reversed on the display as well as scrambled, using TS100 image. Here is an example I did a "flip horizontally" on.

What I see with my eyes:

![display-1](https://user-images.githubusercontent.com/22427598/215286525-e50f67ca-d84b-4e8f-9829-693aff369e33.jpg)

After flipping horizontally on the PC:

![display-2](https://user-images.githubusercontent.com/22427598/215286533-7d1b9bd3-4441-4305-b01f-75c6f14d93cd.jpg)

I tested all the IronOS options, it would only flash the TS based images, and only HEX. Flashing back to Sequre OS worked too. 

| File             | Flashed? |
| ---------------- | -------- |
| SI012_V2.21.hex  | Yes      |
| TS100_EN.hex     | Yes      |
| TS80_EN.hex      | Yes      |
| TS80P_EN.hex     | Yes      |
| TS100_EN.dfu     | No       |
| TS100_EN.bin     | No       |
| Pinecilv2_EN.hex | No       |
| Pinecil_EN.hex   | No       |
| MHP30_EN.hex     | No       |

# My Board

- Board print: SEQURE SI012 V1.4
- Board colour: Black
- Chips: Same as Dmitriy369856

# USB Info 

**Holding (A) at boot**

- Screen shows "UPDATE V6.0 [READY]" text only
- Buttons do nothing
- Mass storage is readable
    - 2.1 MB (1.6 MB free)
    - FAT (12-bit version)

```
[26908.823509] usb 3-4: new full-speed USB device number 6 using xhci_hcd
[26908.973592] usb 3-4: New USB device found, idVendor=0483, idProduct=5720, bcdDevice= 2.00
[26908.973603] usb 3-4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[26908.973608] usb 3-4: Product: STM32 Mass Storage
[26908.973611] usb 3-4: Manufacturer: STMicroelectronics
[26908.973614] usb 3-4: SerialNumber: 005A7A7A4E59
[26908.977568] usb-storage 3-4:1.0: USB Mass Storage device detected
[26908.978065] scsi host1: usb-storage 3-4:1.0
[26909.988670] scsi 1:0:0:0: Direct-Access     SEQURE   UPDATE                PQ: 0 ANSI: 2
[26909.989216] sd 1:0:0:0: Attached scsi generic sg1 type 0
[26909.989848] sd 1:0:0:0: [sdb] 4096 512-byte logical blocks: (2.10 MB/2.00 MiB)
[26909.990069] sd 1:0:0:0: [sdb] Write Protect is off
[26909.990073] sd 1:0:0:0: [sdb] Mode Sense: 03 00 00 00
[26909.990272] sd 1:0:0:0: [sdb] No Caching mode page found
[26909.990275] sd 1:0:0:0: [sdb] Assuming drive cache: write through
[26909.997970]  sdb:
[26909.998268] sd 1:0:0:0: [sdb] Attached SCSI removable disk
```

**Holding (B) at boot**

- Screen shows "LOGO" text only
- Buttons do nothing
- Mass storage is not readable

```
[27020.282403] usb 3-4: USB disconnect, device number 6
[27020.299697] FAT-fs (sdb): unable to read boot sector to mark fs as dirty
[27025.915729] usb 3-4: new full-speed USB device number 7 using xhci_hcd
[27026.066253] usb 3-4: New USB device found, idVendor=0483, idProduct=5720, bcdDevice= 2.00
[27026.066264] usb 3-4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[27026.066268] usb 3-4: Product: STM32 Mass Storage
[27026.066271] usb 3-4: Manufacturer: STMicroelectronics
[27026.066274] usb 3-4: SerialNumber: .(

[27026.068743] usb-storage 3-4:1.0: USB Mass Storage device detected
[27026.069238] scsi host1: usb-storage 3-4:1.0
[27027.076865] scsi 1:0:0:0: Direct-Access     LOGO                           PQ: 0 ANSI: 2
[27027.077434] sd 1:0:0:0: Attached scsi generic sg1 type 0
[27027.077850] sd 1:0:0:0: [sdb] 4096 512-byte logical blocks: (2.10 MB/2.00 MiB)
[27027.078083] sd 1:0:0:0: [sdb] Write Protect is off
[27027.078088] sd 1:0:0:0: [sdb] Mode Sense: 03 00 00 00
[27027.078301] sd 1:0:0:0: [sdb] No Caching mode page found
[27027.078304] sd 1:0:0:0: [sdb] Assuming drive cache: write through
[27027.171277]  sdb:
[27027.171726] sd 1:0:0:0: [sdb] Attached SCSI removable disk

**Reply by Dmitriy369856** [reply]
Manual how to install your logo https://sequremall.com/blogs/frontpage-blog/how-to-customize-the-boot-logo-of-si012-si012-pro
Here is mine ) useless
![Скриншот 28-01-2023 231001](https://user-images.githubusercontent.com/120063968/215288854-0c2a727f-be76-4420-8a87-2eaba3f42514.jpg)

**Comment by mr-sm1th** [comment]
old mcu version:
![old](https://user-images.githubusercontent.com/55600050/206436602-25ed726b-b844-4b0d-bdf4-a153cdfc661b.jpg)

**Reply by rudrakabir** [reply]
any updates on this? Has anyone installed ironos on a si012?

**Comment by t1nkersec** [comment]
Also interested!

**Comment by i0papb** [comment]
hi, 
any idea about that it would work on si012 pro  I have board version  si012pro v1.4 (black)  and it works like the old version hold a then update v6.51 ready with mas storage?

**Comment by jemarro12** [comment]
Si012 Pro V1.5 (PINK PCB)
MCU used is CKS32F103C8T6 (the only one used)

![IMG_20250324_223651837](https://github.com/user-attachments/assets/3c870f85-1002-42c6-b9f2-ac2dac04ce5f)
![IMG_20250324_223700580](https://github.com/user-attachments/assets/c7703c7a-dba9-43f9-a49d-8bb72756d959)
![IMG_20250324_223628225](https://github.com/user-attachments/assets/e954b5e4-29fc-4e0b-8eb5-b1bdabd9a644)
![IMG_20250324_223704218](https://github.com/user-attachments/assets/e347f440-53dc-40ed-9030-4f6a00965abf)
![IMG_20250324_223643885](https://github.com/user-attachments/assets/e0468c92-0bf5-4820-98c7-ab118700409c)
